### ISSUE TITLE
Support external OAuth login

### ISSUE THREAD

**Issue by lowgos** [Chocobozzz/PeerTube#1607]
Hi everyone,

First issue here. More of a question about PeerTube features and privacy (btw love the project ❤️!)

I would like to raise a question, let me make an example up

## Illustration
Say we are a school, we need to host some "private" videos. Like courses, news, communication, general assemblies of student associations, etc.
In our school we have an OAuth Service already set up with our intranet.

In my example we would like to :

- Make our students login via our OAuth
- Restrict some videos to logged in users only, like courses or assemblies
- Let foreign users watch other content too, maybe communication videos, content from students, etc

I guess it sounds a bit different from the idea that PeerTube is opened to everyone but... is it doable ? Are there features that allows me to do that easily ?

## Solutions, maybe ?
In any case, I know I could fork the project and then adapt it to my needs, but I think any company would be glad to use PeerTube to answer this common case about internal communication without being opened to the world. 

## Openings
Maybe federate instances from every sub-company which already have an instance etc. ?

## Conclusion
I couldn't find the exact information somewhere in the Documentation or in the issues (I found a related issue: #520) so I'm asking here since this repository is really active and issues are welcome 😇 

# tl;dr
a bit like #520 but for companies, schools, etc. Semi-closed instances and with OAuth maybe.

Hope I didn't make you waste time.
Love.

lowgos

**Comment by Chocobozzz** [comment]
It's an interesting way to use PeerTube. But I don't see differences with https://github.com/Chocobozzz/PeerTube/issues/520

I think we could add an *internal* privacy setting to videos, like on gitlab for example (ie only available for logged-in users).

**Comment by lowgos** [comment]
OAuth is the only difference. Sorry if my issue is too much of a duplicate. I think it's my first issue ever 😇

I work at 42 Paris, so if people there are hyped about PeerTube we could meet with you and Framasoft. Could be a good opportunity.

Anyway. Thank you for taking my opinion into account. Wish you the best !

lowgos

**Comment by Chocobozzz** [comment]
Can now be implemented using auth plugins like we did for [openid](https://framagit.org/framasoft/peertube/official-plugins/-/tree/master/peertube-plugin-auth-openid-connect) for example
